pkgs/test_core/CHANGELOG.md

```diff
@@ -4,6 +4,8 @@
 * Report incomplete tests as errors in the JSON reporter when the run is
   canceled early.
 * Don't log the --test-randomization-ordering-seed if using the json reporter.
+* Add a new exit code, 79, which is used when no tests were ran.
```

Comment:
This isn't quite accurate, it will be a different code if you run a suite with no tests at all.

We could expand this sentence

"which is used when no tests were ran due to tags or other filters matching no tests."

Reply:
This new exit code is used when:

- No tests were ran and:
  - No tags were configured
  - No name patterns were provided
  
In the other cases the exit code 65 is used (no change for the pattern case, new message/code for the tag case)

Reply:
I do think its confusing though and am leaning towards making these all the same exception/exit code (79).

Reply:
> am leaning towards making these all the same exception/exit code (79).

I think I'd prefer that as well.

Reply:
Although I think handling the no tests in suite case might be trickier. I think that's exit code 1 today, same as a test failure.

Reply:
> Although I think handling the no tests in suite case might be trickier. I think that's exit code 1 today, same as a test failure.

Correct, it is exit code 1 today. With this PR it changes to exit code 79.

Reply:
> With this PR it changes to exit code 79.

How?

No tests defined is handled by https://github.com/jepperaskdk/test/blob/76cf9b7744eda85e27e45369fe63d13b7f14f56c/pkgs/test_core/lib/src/runner.dart#L149 which doesn't throw this exception.

Reply:
Oh, nevermind, I see now the removal of `_config.suiteDefaults.patterns.isNotEmpty)`